analysis.md

```diff
@@ -0,0 +1,79 @@
+
+In this article, I'll go through the results of my analysis and talk about the methods I used and thought processes at each step.
+
+## Extracting the Source Code
+I used [oletools](https://github.com/decalage2/oletools) to extract the VBA source code from the macro. After installing oletools, simply run the command `olevba <filepath>` to reveal the macro.
+
+## Layer 1 - VB and Embedded PowerShell
+Right off the bat there are several things the attacker does to make the script difficult to unravel. The author embeds a lot of newline spacing, widley spreading apart contents to make important code harder to spot. After revealing the macro with oletools, you'll probably have to do a _lot_ of scrolling to read the whole script. Even worse, if the macro is revealed through a terminal with a limited scrollback buffer, part of the script would be hidden.
+
+There is a function `ParsingA()` that appears to download content from the Internet, but it never seems to be used. This is just an attempt to redirect attention to something unimportant and distract from the actual payload.
+
+You'll also notice that the author uses gratuitous string concatenation to break simple string signatures. This helps the attacker evade malware analysis tools that may look for strings like `"winmgmts:\\.\root\cimv2"` and be thrown off by something like 
+```
+"w" & "" & "in" & "" & "mgm" & "" & "ts" & "" & ":" & "" & "\\" & "." & "\r" & "" & "oot\c" & "" & "imv" & "" & "2"
+```
+
+Also notice the large Base64 blob that `lStr` contains. This blob is decoded and decompressed it for execution. Let's dive further.
+
+### Unraveling the Base64
+Below is a simple Python3 script I use to decode the Base64 blob mentioned earlier.
+
+```
+import base64
+import sys
+
+ifile = open(sys.argv[1], "r")
+
+encoded = ifile.read()
+
+sys.stdout.write(base64.b64decode(encoded))
+```
+After decoding, you'll notice that the resulting contents are still unreadable. We can tell through a few ways that the contents are in GZIP format. Most obviously, the VB string stored in `lstr` has a command to decompress the decoded contents using a `System.IO.Compression.GZipStream` object. Another way we can identify this as a gzip file is through its [magic number](https://en.wikipedia.org/wiki/List_of_file_signatures), the beginning file signature (1F8B). We can also use the `file <filepath>` command on a Unix machine to recognize the data.
+
+Now we'll decompress the contents using `gunzip <filepath>`, which creates a decompressed file with the same name. Opening it will reveal obfuscated PowerShell code with strange variable names. After cleaning up this mess and refactoring the code, there are some interesting things to note about malware persistence. This is covered in the following section.
+
+## Layer 2
```

Comment:
Try giving each layer a short title to remind readers what the layers are. Without some kind of anchor it's easy to lose track since they're just slightly arbitrary. Like, layer 1 could be VB, layer 2 could be base64 decoded gzip blob, layer 3 could be decompressed blob. It's fine the way you have them numbered now, just label them a bit.